homeassistant/components/mqtt/mixins.py

```diff
@@ -223,6 +225,29 @@ def validate_device_has_at_least_one_identifier(value: ConfigType) -> ConfigType
 )
 
 
+def warn_for_legacy_schema(domain: str) -> Callable:
+    """Warn once when a legacy platform schema is used."""
+    warned = set()
+
+    def validator(config: ConfigType) -> ConfigType:
+        """Return a validator."""
+        nonlocal warned
+
+        if domain in warned:
+            return config
+
+        _LOGGER.warning(
+            "Manually configured MQTT item(s) found under platform key '%s'. Manually configured MQTT item configurations have been moved to the integration key, please move the configuration of MQTT %s to mqtt->%s",
```

Comment:
It will be:
> Manually configured MQTT item(s) found under platform key 'fan'. Manually configured MQTT item configurations have been moved to the integration key, please move the configuration of MQTT fan to mqtt->fan

Maybe this would be better:
> Manually configured MQTT fan(s) found under platform key 'fan', please move to the mqtt integration key mqtt->fan

Reply:
I'd remove the `mqtt->fan`. That's not how the YAML looks and it's hard to interpret what it means. Maybe just link to the docs?

Reply:
About the long string, what's the state of Black's preview of string formatting: https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#improved-string-processing ?

Reply:
Something like this?
```python
        _LOGGER.warning(
            "Manually configured MQTT fan(s) found under platform key '%s', "
            "please move to the mqtt integration key, see https://www.home-assistant.io/integrations/%s.mqtt/",
            domain,
            domain,
        )
```

Reply:
The diff looks ok at a glance:
```
black --fast --preview homeassistant
```